### ISSUE TITLE
Unable to make face unlock working on Cepheus

### ISSUE THREAD

**Issue by finestedm** [PixelExperience/android-issues#113]
## Build date

```
PixelExperience_Plus_cepheus-11.0-20210514-0318-OFFICIAL.zip
```

## Expected Behavior

```
Face unlock working
```

## Current Behavior

```
Face unlock not even trying to unlock. No indication of it trying to search for a face. It sets up without problems. It also works in auto fill or banking apps after pressing "face id" button. I haven't tested if it also doesn't work in normal version as this is what happened in plus version.
```

## Possible Solution

```
No idea
```

## Steps to Reproduce

```
1. Set up fingerprint
2. Set up face unlock
3. Lock device
4. Turn on screen
```

/device cepheus
/version eleven_plus

**Comment by jjpprrrr** [comment]
Face unlock works fine here on my cepheus.

**Comment by finestedm** [comment]
> Face unlock works fine here on my cepheus.

Is it working on plus version? I just flashed everithing obce again and its still bot working :(

**Comment by finestedm** [comment]
https://youtu.be/360bcoP6KuI

Here is a video showing a problem. Face is remembered, I lock the screen and it doesn't want to unlock with face recognition after turning the screen on.

**Comment by jjpprrrr** [comment]
> > Face unlock works fine here on my cepheus.
> 
> Is it working on plus version? I just flashed everithing obce again and its still bot working :(

Yes I am using the plus version.

**Comment by finestedm** [comment]
Please find attached 2 logcat files. 
One is from setting up camera unlock, second one is made after face unlock is setup, closing screen, pressing power up button and trying to unlock with face (still not working).
[Trying to unlock.txt](https://github.com/PixelExperience/android-issues/files/6566392/Trying.to.unlock.txt)
[Setup face unlock.txt](https://github.com/PixelExperience/android-issues/files/6566393/Setup.face.unlock.txt)

**Comment by jhenrique09** [comment]
Are you using any magisk module? Or any flashable zip? Please check that.

Also, you tried clean flash?

**Comment by EcrosoftXiao** [comment]
> Please find attached 2 logcat files.
> One is from setting up camera unlock, second one is made after face unlock is setup, closing screen, pressing power up button and trying to unlock with face (still not working).
> [Trying to unlock.txt](https://github.com/PixelExperience/android-issues/files/6566392/Trying.to.unlock.txt)
> [Setup face unlock.txt](https://github.com/PixelExperience/android-issues/files/6566393/Setup.face.unlock.txt)

It works fine here anyway. Try clean install and re-set up face unlock.

**Comment by finestedm** [comment]
No, totally clean install, no other files, root or magisk. Tried resetting, setting up face unlock countless times. I also tried PixelPlus ROM and it also not working but it works just fine on Fluid ROM. Really strange...

**Comment by indc9** [comment]
Face unlock works fine on my cepheus.
Some times missunderstand the face and dont works, but in our device nice light conditions is mandatory to work properly.
Try to reset the face settings in the best possible light conditions and try different distances between the face and the mobile during the process.

**Comment by finestedm** [comment]
> Face unlock works fine on my cepheus.
> Sometimes missunderstand the face and dont works, but in our device nice light conditions is mandatory to work properly.
> Try to reset the face settings in the best possible light conditions and try different distances between the face and the mobile during the process.

Like I said before the problem is not that the phone doesn't recognize the face, it doesn't initate the camera unlocking in the first place. At this point I think I tried everything. I checked of face unlocking system app has permissions but it's unchangeable so no problem there neither. Hardware can't be wrong as it's part of the ROM zip file. I really don't know what else can be a problem :(

**Comment by finestedm** [comment]
Couple of ideas that I got as I keep thinking what might be a problem for me:

> 1. I have a different hardware - maybe Xiaomi uses different components for different distributions to different countries

>2. I need to try different recovery. I used newest twrp.

> 3. I previously used Xiaomi.eu ROM instead of stock ROM. Maybe it left something in the memory even tho I wiped the device completely before flashing this ROM. 

I checked logcat to see if I can find something suspicious but I'm not an expert. Do you see anything indicating a problem?
Thanks.

**Comment by jjpprrrr** [comment]
> Couple of ideas that I got as I keep thinking what might be a problem for me:
> 
> > 1. I have a different hardware - maybe Xiaomi uses different components for different distributions to different countries
> 
> > 1. I need to try different recovery. I used newest twrp.
> 
> > 1. I previously used Xiaomi.eu ROM instead of stock ROM. Maybe it left something in the memory even tho I wiped the device completely before flashing this ROM.
> 
> I checked logcat to see if I can find something suspicious but I'm not an expert. Do you see anything indicating a problem?
> Thanks.

TBH I didn't see anything wrong in logs.

**Comment by finestedm** [comment]
> > Couple of ideas that I got as I keep thinking what might be a problem for me:
> > > 1. I have a different hardware - maybe Xiaomi uses different components for different distributions to different countries
> > 
> > 
> > > 1. I need to try different recovery. I used newest twrp.
> > 
> > 
> > > 1. I previously used Xiaomi.eu ROM instead of stock ROM. Maybe it left something in the memory even tho I wiped the device completely before flashing this ROM.
> > 
> > 
> > I checked logcat to see if I can find something suspicious but I'm not an expert. Do you see anything indicating a problem?
> > Thanks.
> 
> TBH I didn't see anything wrong in logs.

Could you or anyone post logcat from face unlocking? I want to compare and search for the problem.
I see in my logs that face unlock isn't doing anything when I turn on the screen. Like it's not active or something...

**Comment by finestedm** [comment]
![Screenshot_20210603-202130_Log_Master](https://user-images.githubusercontent.com/85036437/120693796-c21b0580-c4a9-11eb-8b75-4268ca7f210e.png)


> > > > Couple of ideas that I got as I keep thinking what might be a problem for me:
> > > > > 1. I have a different hardware - maybe Xiaomi uses different components for different distributions to different countries
> > > > 
> > > > 
> > > > > 1. I need to try different recovery. I used newest twrp.
> > > > 
> > > > 
> > > > > 1. I previously used Xiaomi.eu ROM instead of stock ROM. Maybe it left something in the memory even tho I wiped the device completely before flashing this ROM.
> > > > 
> > > > 
> > > > I checked logcat to see if I can find something suspicious but I'm not an expert. Do you see anything indicating a problem?
> > > > Thanks.
> > > 
> > > 
> > > TBH I didn't see anything wrong in logs.
> > 
> > 
> > Could you or anyone post logcat from face unlocking? I want to compare and search for the problem.
> > I see in my logs that face unlock isn't doing anything when I turn on the screen. Like it's not active or something...
> 
> [0603_pe_plus_cepheus_face_unlock.log](https://github.com/PixelExperience/android-issues/files/6592087/0603_pe_plus_cepheus_face_unlock.log)

Ok, thanks a lot! In attached screenshot I see the beginning of what's interesting. The FaceService doesn't show up in my logs. No idea why but everything that happens next for you, starting with this log, doesn't happen in my case 😔

**Comment by finestedm** [comment]
Hello. I wanted to share with you my discovery I made. I don't think it will be any useful as I probably was the only one who got the issue with face unlock but anyway. I decided to wipe the phone clean, install PE and just skip setting up the phone. I went to the settings, set up face unlock and voila, it works! So I decided to factory reset the phone, set it up and restore data and then set up face unlock. It stopped working. I even tried first setting up the face unlock and then restoring data but the moment I did the face unlock just disables. Turns out there has to be something stored in Google's data that interferes with the ROM. Maybe some security settings that got saves on previous ROM or on stock miui. Who knows? 
Anyway, face unlock works like a charm, it's the last thing I was missing and now the ROM is complete. Thank you!
